mne/io/nicolet/nicolet.py

```diff
@@ -0,0 +1,212 @@
+        ECG channels. Values should correspond to the electrodes in the
+        data file. Default is None.
+    emg : list or tuple
+        Names of channels or list of indices that should be designated
+        EMG channels. If None (default), the channel names beginning with
+        ``EMG`` are used.
+    misc : list or tuple
+        Names of channels or list of indices that should be designated
+        MISC channels. Values should correspond to the electrodes in the
+        data file. Default is None.
+    preload : bool or str (default False)
+        Preload data into memory for data manipulation and faster indexing.
+        If True, the data will be preloaded into memory (fast, requires
+        large amount of memory). If preload is a string, preload is the
+        file name of a memory-mapped file which is used to store the data
+        on the hard drive (slower, requires less memory).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    def __init__(self, input_fname, montage=None, eog=None, ecg=None, emg=None,
+                 misc=None, preload=False, verbose=None):
+        input_fname = path.abspath(input_fname)
+        info, header_info = _get_nicolet_info(input_fname, eog, ecg, emg, misc)
+        last_samps = [header_info['num_samples'] - 1]
+        _check_update_montage(info, montage)
+        super(RawNicolet, self).__init__(
+            info, preload, filenames=[input_fname], raw_extras=[header_info],
+            last_samps=last_samps, orig_format='int',
+            verbose=verbose)
+
+    def _read_segment_file(self, data, idx, offset, fi, start, stop, cals,
+                           mult):
+        """Read a chunk of raw data"""
+        if mult is not None:
+            raise NotImplementedError()
+        nchan = self.info['nchan']
+        sel = np.arange(nchan)[idx]
+        cal = np.array([ch['cal'] for ch in self.info['chs']])
+
+        data_offset = self.info['nchan'] * start * 2
+        data_left = (stop - start + 1) * nchan
+        # Read up to 100 MB of data at a time.
+        blk_size = min(data_left, (50000000 // nchan) * nchan)
+
+        with open(self._filenames[fi], 'rb', buffering=0) as fid:
+            fid.seek(data_offset)
+            # extract data in chunks
+            for blk_start in np.arange(0, data_left, blk_size) // nchan:
+                blk_size = min(blk_size, data_left - blk_start * nchan)
+                block = np.fromfile(fid, '<i2', blk_size)
+                block = block.reshape(nchan, len(block) // nchan,
+                                      order='F')[sel].astype(float)
+                blk_stop = blk_start + block.shape[1]
+                data[:, blk_start:blk_stop][sel] = block[sel] *\
+                    np.expand_dims(cal[sel], axis=1)
```

Comment:
Yeah it was working on my machine too, but travis was still failing.

Basically the ``idx`` is an unused param. The data is always extracted from all the channels available. That makes the check unnecessary, but indeed, the second pick is not needed.